### PR TITLE
Add property attribute

### PR DESCRIPTION
**Pull request by csnewman**
Adds a property attribute that can be used to override the auto generated settings.

Example:
```
        [Property(Name = "Test", Access = PropertyAccess.Read, Type = typeof(int))]
        public string Something;

        [Property(Name = "Test2")]
        public string Something2;

        [Property(Access = PropertyAccess.Read)]
        public string Something3;
```

### CODE REVIEW

File: src/Tmds.DBus/PropertyAccess.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright 2016 Tom Deseyn <tom.deseyn@gmail.com>
+// This software is made available under the MIT License
+// See COPYING for details
+
+namespace Tmds.DBus.CodeGen
```
Comment (tmds) [inline src/Tmds.DBus/PropertyAccess.cs]: Update namespace -> `Tmds.DBus`

Comment (csnewman) [reply src/Tmds.DBus/PropertyAccess.cs]: Whoops, not sure how I missed that!

File: src/Tmds.DBus/PropertyAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        /// <summary>
+        /// Specifies the mutability of the property.
+        /// </summary>
+        public PropertyAccess Access { get; set; } = PropertyAccess.ReadWrite;
+
+        /// <summary>
+        /// If not null, used to override the type used in the signature.
+        /// </summary>
+        public Type Type { get; set; }
```
Comment (tmds) [inline src/Tmds.DBus/PropertyAttribute.cs]: We need to (de)serialize the field type, so I don't think a user should override this.

**Comment by tmds** [comment]
Can you update the test? You can add some new fields with the PropertyAttribute to this:

https://github.com/tmds/Tmds.DBus/blob/c173ac56daf611a103050c0e465e89bfa15eef7b/test/Tmds.DBus.Tests/IntrospectionTests.cs#L16-L23

and update the expected xml.

File: src/Tmds.DBus/PropertyAccess.cs (hunk 1 above)
Comment (csnewman) [reply src/Tmds.DBus/PropertyAccess.cs]: Fixed

File: src/Tmds.DBus/PropertyAttribute.cs (hunk 2 above)
Comment (csnewman) [reply src/Tmds.DBus/PropertyAttribute.cs]: Removed.

**Comment by csnewman** [comment]
I've added the property to some extra fields in the unit test and updated the xml. They pass.

**Comment by csnewman** [comment]
I have also cleaned up the code to match the code style of the rest of the project

**Comment by tmds** [comment]
LGTM, Thanks!

I'll publish a new release on nuget.org next week.

**Comment by tmds** [comment]
@csnewman fyi, I've published 0.7.0 on nuget.org which includes your recent contributions!

**Pull request merged**